### ISSUE TITLE
Stile Type Quest improvements

### ISSUE THREAD

**Issue by 10992-osm** [streetcomplete/StreetComplete#2991]
When I was adding the stile type with StreetComplete, I found myself adding a note for stone stepover stiles, since they obviously don't look like wooden ones, as the whole setup is rather different.  [On the wiki](https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dstile#Examples), it says that they are, in fact, stepover stiles, and should be tagged as such (with `material=stone`).  Two pictures might help, showing both a wooden and stone stepover stile, or another option could be added which also tags `material=stone`.

There appear to be a handful of examples of stiles tagged `stile=stone_steps` or `stile=stone steps`, which I imagine are the same as the stiles I found that, according to the wiki, are meant to be tagged `stile=stepover`.

As well as this (though I might open another issue if it is deemed appropriate), a new quest could be created asking for the material of stiles.  Currently, [according to Taginfo](https://taginfo.openstreetmap.org/tags/?key=barrier&value=stile#combinations), 16.9% of `barrier=stile` objects are tagged with `material=*`.  58.01% of [objects tagged with `stile=*`](https://taginfo.openstreetmap.org/keys/stile#combinations) are tagged with `material=*`.

An example note that I made is [note 2723843](https://osm.org/note/2723843).

**Comment by matkoniecz** [comment]
> I found myself adding a note for stone stepover stiles, since they obviously don't look like wooden ones, as the whole setup is rather different.

Oh, I see how it may happen.

Sadly https://wiki.openstreetmap.org/wiki/File:Stone_stepover.jpg cannot be used as it is missing a copyright info (it also should be replaced on the OSM Wiki). Also, it is quite noisy anyway and would likely be hard to crop it into a square image.

Can you find some alternative image, maybe by looking through stile photos on Geograph?

https://en.wikipedia.org/wiki/File:Squeezer_stile._-_geograph.org.uk_-_110502.jpg

https://commons.wikimedia.org/wiki/File:Family_of_ladder_stiles_-_geograph.org.uk_-_42040.jpg

https://commons.wikimedia.org/wiki/File:Daffodil_Way,_Dymock_-_geograph.org.uk_-_69563.jpg

all are from Geograph.

Finding a good match with a clear view, without visual noise, fitting square crop is quite time consuming so help would be appreciated.

> As well as this (though I might open another issue if it is deemed appropriate), a new quest could be created asking for the material of stiles

Is this data used or potentially useful? (beyond "super detailed 3D renderings")

> When I was adding the stile type with StreetComplete

Glad to see that this quest is being used :) (#2766)

**Comment by 10992-osm** [comment]
> Sadly https://wiki.openstreetmap.org/wiki/File:Stone_stepover.jpg cannot be used as it is missing a copyright info (it also should be replaced on the OSM Wiki). Also, it is quite noisy anyway and would likely be hard to crop it into a square image.
> 
> Can you find some alternative image, maybe by looking through stile photos on Geograph?
> 
> https://en.wikipedia.org/wiki/File:Squeezer_stile._-_geograph.org.uk_-_110502.jpg
> 
> https://commons.wikimedia.org/wiki/File:Family_of_ladder_stiles_-_geograph.org.uk_-_42040.jpg
> 
> https://commons.wikimedia.org/wiki/File:Daffodil_Way,_Dymock_-_geograph.org.uk_-_69563.jpg
> 
> all are from Geograph.
> 
> Finding a good match with a clear view, without visual noise, fitting square crop is quite time consuming so help would be appreciated.

I'll have a look and add another comment if I find anything.

> Is this data used or potentially useful? (beyond "super detailed 3D renderings")

No idea, beyond the fact that `barrier=stile` + `stile=stepover` + `material=wood` is fairly different from `barrier=stile` + `stile=stepover` + `material=stone`, hence this issue.  The method of traversing them is slightly different, since you tend to have to climb up onto the stone wall, rather than swinging your legs over a wooden fence.

**Comment by 10992-osm** [comment]
### Possible reasons for tagging the material of stiles
Although the wiki is of course not gospel, it does give some reasons why you might want to tag the stile material, beyond what I mentioned above.

[The wiki page for `barrier=stile`](https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dstile#Stile_type) says:

>Stone stiles are of particular interest as they form part of the historic landscape. Identifying them helps with preservation work.

>If known, the material should also be tagged. For example, stone stiles form part of the historic landscape character of the UK, and identifying their location is useful in preservation work. Where possible, the stile should be tagged with material=* with values stone, wood, or metal.

### Possible photos of stone stepover stiles

I think the first one is probably the best.

1. https://www.geograph.org.uk/photo/152706 (fairly typical, I've cropped this one to a square and it still seems good, see below)
![Stone stepover stile and metal gate next to main road with a footpath sign](https://user-images.githubusercontent.com/83511672/122989019-40c4dd80-d39a-11eb-8ae3-0e2f1a4be1b2.jpg)
2. https://commons.wikimedia.org/wiki/File:Dry_Stone_Wall_and_Stile,_Yorkshire_Dales_(3875164717).jpg (fairly typical, they often have a gate on one side, but not all of them)
3. https://www.geograph.org.uk/photo/555591 (again, fairly typical)
4. https://commons.wikimedia.org/wiki/File:Drystone_Wall_(26721799888).png (the wooden part of fence often isn't present, and the angle of this photo may be less than ideal)
5. https://www.geograph.org.uk/photo/398328 (fairly typical, though the angle makes the steps a little difficult to distinguish)
6. https://www.geograph.org.uk/photo/2225748 (steps are easy to see, but in my opinion the design is less typical as there is no gap in the top of the wall)
7. https://www.geograph.org.uk/photo/3634198 (reasonable photo, steps are not as clear as they might be)
8. https://www.flickr.com/photos/atoach/4601787397 (reasonable photo)
9. ![Image of a stone stepover stile](https://user-images.githubusercontent.com/83511672/122988121-4968e400-d399-11eb-937c-51c6463fa5b5.jpg) (my image, probably a bit crowded by vegetation, if suitable I can give it a suitable licence and upload it to Wikimedia Commons).

**Comment by matkoniecz** [comment]
Thank you for the photos! I will need to check it in app itself but some are very promising.

**Comment by matkoniecz** [comment]
I asked on talk-gb for confirmation before editing wiki (I will replace photo problematic due to copyright) and implementing it, as my experience with stile tagging is exactly zero.

**Comment by 10992-osm** [comment]
I just thought I'd summarise the responses received to @matkoniecz's messages on [[Talk-GB]](https://lists.openstreetmap.org/listinfo/talk-gb "[Talk-GB] OpenStreetMap mailing list") in [June 2021](https://lists.openstreetmap.org/pipermail/talk-gb/2021-June/ "Archive for June 2021 on the [Talk-GB] OpenStreetMap mailing list"), so that they are collected in one place, and summarised.

# [Tagging `material=*` on stiles](https://lists.openstreetmap.org/pipermail/talk-gb/2021-June/027238.html)

[**Mike Collinson**](https://lists.openstreetmap.org/pipermail/talk-gb/2021-June/027240.html) thinks that:
- stiles *can be tagged with `material=*`*, rather than *should be tagged*.
- *stone stiles are generally much easier to cross in wet weather as wood gets slippy*
- the tags `style_type=*`, `step_count` (see #3141) and [`dog_gate=*`](https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dstile#Dog_gate) are the most useful tags for stiles
- tagging `material=stone` *adds a historical dimension*, but isn't ***generally** useful*.

[**Jerry (SK53)**](https://lists.openstreetmap.org/pipermail/talk-gb/2021-June/027244.html) thinks that:
- it *is useful contextual information*

[**Tom Crocker**](https://lists.openstreetmap.org/pipermail/talk-gb/2021-June/027245.html) thinks that:
- it's *not very unusual for stiles to be added where the wall or (more so)
fence exists but isn't in the database* (which means that you can necessarily infer the material of a stile from what it intersects)
- it *may as well be added by people who want to*.

[**Tony Shield**](https://lists.openstreetmap.org/pipermail/talk-gb/2021-June/027248.html) thinks that:
- it *is useful particularly in urban situations*, citing that *steel squeezer designed to prevent motorcyclists is likely to be there several years after installation, whereas a wooden one will likely have been destroyed to allow access* (for a case in point see [OSM Note 2647657](https://osm.org/note/2647657))
- *metal squeezers are also difficult to navigate with a bicycle and tall people and those with bulky rucsac's* (sic)

[**Philip Barnes**](https://lists.openstreetmap.org/pipermail/talk-gb/2021-June/027247.html) thinks that:
- a *useful streetcomplete question that is going to occur more often is it is not a stile anymore and whether it has been replaced by a gap, a gate or kissing_gate*

**Comment by matkoniecz** [comment]
Thanks again for help! That was exactly what I was blocked on (and seeing someone putting work into something is REALLY motivating to also finish it).

So conclusion is to provide option to select stone/wood (if there are many where there is substantial use of both it can be also handled somehow) as extra complexity and effort is minimal AND it reduces confusion reported here.

------

https://www.flickr.com/photos/atoach/4601787397 was tempting but noise in opening made it hard to understand when rescaled

https://www.geograph.org.uk/photo/3634198 seems very fitting

![screen01](https://user-images.githubusercontent.com/899988/129688755-f71992b3-3031-425f-9e9e-b49681b4710c.png)
![screen](https://user-images.githubusercontent.com/899988/129688762-6300d8b4-745d-4ac9-aa39-31475cfa73d9.png)

**Comment by 10992-osm** [comment]
This all looks very good.

Was a decision reached on the general tagging of `material=*` on stiles, or is it just in this particular circumstance that StreetComplete will tag it.  I personally think that this is where it's most useful (on `stile=stepover`).

---

In relation to [your comment](https://github.com/streetcomplete/StreetComplete/pull/3175#issuecomment-900118853) on #3175:

> I can be easily convinced to use https://www.geograph.org.uk/photo/398328 or https://commons.wikimedia.org/wiki/File:Dry_Stone_Wall_and_Stile,_Yorkshire_Dales_(3875164717).jpg instead - to show that small piece of wood is not changing that it is a stone stile.

I'm not sure that https://www.geograph.org.uk/photo/398328 is particularly clear, so if I were to choose between those two, I'd choose https://commons.wikimedia.org/wiki/File:Dry_Stone_Wall_and_Stile,_Yorkshire_Dales_(3875164717).jpg.  The one you've already got looks good, though, as you say, the other two demonstrate that the addition of a small wooden gate does not change the type of stile

---

I also have a few further points.  I have some others which I might add to a separate issue later, as they are less closely related to this issue.

Firstly, with regard to handling existing `barrier=stile`+`stile=stepover` without `material=*`, does (or should) StreetComplete:
1. ignore them, leaving the `material=*` tag to be added manually by any editor who wishes to do so; or
2. treat them as having missing tags, and ask the full stile-type quest; or
3. ask a more simplified quest (either asking just for the material, or asking is this a wooden step-over or stone step-over)?

Secondly, what mechanism is (or should be) provided in case a step-over stile uses another material?  You can get metal stiles, but these are quite rare, so they need not be a main option, but they could be hidden: e.g. `Other answers` → `This is a step-over stile made of another material`.

**Comment by matkoniecz** [comment]
> ignore them, leaving the material=* tag to be added manually by any editor who wishes to do so; or

I went with this one so far given responses on talk-gb, but I can be easily convinced to ask question also for them.

**Comment by 10992-osm** [comment]
> > ignore them, leaving the material=* tag to be added manually by any editor who wishes to do so; or
> 
> I went with this one so far given responses on talk-gb, but I can be easily convinced to ask question also for them.

I'm not sure really.  The other points I had are below, and with the first one, if stiles are re-surveyed, then one could just wait until the re-survey until the material is asked (one could have a shorter resurvey time if no `material=*` tag is present).

- As far as I can tell, stiles are never resurveyed.  I agree with what Philip Barnes' said ([above](https://github.com/streetcomplete/StreetComplete/issues/2991#issuecomment-899841013)) about resurveying stiles on regular intervals.  I'm not really sure what the interval should be, but my starting suggestion would be 5 years.  In the UK, stiles are gradually being replaced in many areas, and they are not supposed to be installed where one did not exist before (I'm not sure at this precise moment whether this is a legal requirement, or just guidance, it might say in one of the links in my comment on #3141).  I'm not sure whether this would be worth asking the whole stile type quest again, or just asking "Is this ladder/wooden step-over/stone step-over/step-over/squeeze stile still here?" (`Yes`, `No, another type of stile`, `No a gate`, `Can't say`).

- When asking the stile quest, and possibly also when re-surveying as above, the option to say that it has been converted into a kissing gate is provided.  Philip Barnes suggested that it might also be turned into gap or other gate.  I think kissing gate is probably the most common, but perhaps the others could also be provided.  Either as principal options or hidden (`Other answers...` → `Another type of gate`/`A gap in the fence/wall/barrier`).

- Mike Collinson mentions dog gates ([`dog_gate=*`](https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dstile#Dog_gate)).  I'm not particularly sure how useful tagging them is, since I never walk with dogs, so don't know how much impact dog gates have.  I'm also not sure how easy the quest would be to define, since some people might interpret anything that allows a dog through (such as a hole in the fence) as a dog gate.  This quest would require consultation with other people, and possibly another discussion on [Talk-GB].

I can move some/all of these points to new (a) issue(s), if necessary.

**Comment by matkoniecz** [comment]
> I can move some/all of these points to new (a) issue(s), if necessary.

dog gates seems to have no support (from older discussion) but stile resurvey makes sense if they keep disappearing - feel free to open a new issue for that
